From 1d1c2d3426f6a69cfc914c49f6531c8012293025 Mon Sep 17 00:00:00 2001
From: Gustavo Ornaghi Antunes <gustavo.ornaghiantunes@windriver.com>
Date: Thu, 5 Dec 2024 09:44:06 -0300
Subject: [PATCH] csi-powerstore: enabling csm-replication

This patch consists of changing the format of the values in templates and
changing the values.yaml for the user be able to enable csm-replication.

Signed-off-by: Gustavo Ornaghi Antunes <gustavo.ornaghiantunes@windriver.com>
---
 charts/csi-powerstore/templates/controller.yaml | 6 +++---
 charts/csi-powerstore/values.yaml               | 1 +
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/charts/csi-powerstore/templates/controller.yaml b/charts/csi-powerstore/templates/controller.yaml
index af93e1b..6c88b58 100644
--- a/charts/csi-powerstore/templates/controller.yaml
+++ b/charts/csi-powerstore/templates/controller.yaml
@@ -114,7 +114,7 @@ rules:
     verbs: ["update", "patch"]
   # below for dell-csi-replicator
   {{- if hasKey .Values.controller "replication" }}
-  {{- if eq .Values.controller.replication.enabled true}}
+  {{- if or .Values.controller.replication.enabled .Values.csm.replication }}
   - apiGroups: ["replication.storage.dell.com"]
     resources: ["dellcsireplicationgroups"]
     verbs: ["create", "delete", "get", "list", "patch", "update", "watch"]
@@ -320,7 +320,7 @@ spec:
         {{- end }}
         {{- end }}
         {{- if hasKey .Values.controller "replication" }}
-        {{- if eq .Values.controller.replication.enabled true}}
+        {{- if or .Values.controller.replication.enabled .Values.csm.replication }}
         - name: dell-csi-replicator
           image: {{ required "Must provide the Dell CSI Replicator image." .Values.controller.replication.image }}
           imagePullPolicy: {{ .Values.imagePullPolicy }}
@@ -424,7 +424,7 @@ spec:
             - name: X_CSI_PODMON_API_PORT
               value: "{{ .Values.podmonAPIPort }}"
             {{- if hasKey .Values.controller "replication" }}
-            {{- if eq .Values.controller.replication.enabled true}}
+            {{- if or .Values.controller.replication.enabled .Values.csm.replication }}
             - name: X_CSI_REPLICATION_CONTEXT_PREFIX
               value: {{ .Values.controller.replication.replicationContextPrefix | default "powerstore"}}
             - name: X_CSI_REPLICATION_PREFIX
diff --git a/charts/csi-powerstore/values.yaml b/charts/csi-powerstore/values.yaml
index 601e878..2801443 100644
--- a/charts/csi-powerstore/values.yaml
+++ b/charts/csi-powerstore/values.yaml
@@ -19,6 +19,7 @@
 
 # CSM Control
 csm:
+  replication: false
   resiliency: false
 
 # driverName: defines the name of driver
-- 
2.34.1
